pkg/controllers/manila/openstack.go

```diff
@@ -55,6 +56,11 @@ func (o *openStackClient) GetShareTypes() ([]sharetypes.ShareType, error) {
 		return nil, fmt.Errorf("failed to create a provider client: %w", err)
 	}
 
+	// we represent version using commits since we don't tag releases
+	ua := gophercloud.UserAgent{}
+	ua.Prepend(fmt.Sprintf("csi-driver-manila-operator/%s", version.Get().GitCommit))
```

Comment:
Have you tested this? This value needs to be set explicitly in the build process by passing `-ldflags`. At first glance I can't see where it would be set. If it is not set explicitly, it would default to the empty string.

Reply:
Yes, the `build` target is provided by https://github.com/openshift/build-machinery-go which we bring in via [an `include`](https://github.com/openshift/csi-driver-manila-operator/blob/4b309d3b1ed31c5d106801933c0174518cf883aa/Makefile#L4-L9). If you run `make build` you'll see the flags provided.

```
❯ make build
fatal: No names found, cannot describe anything.
go build -mod=vendor -trimpath -ldflags "-X github.com/openshift/csi-driver-manila-operator/pkg/version.versionFromGit="v0.0.0-unknown-babb5f0a" -X github.com/openshift/csi-driver-manila-operator/pkg/version.commitFromGit="babb5f0a" -X github.com/openshift/csi-driver-manila-operator/pkg/version.gitTreeState="clean" -X github.com/openshift/csi-driver-manila-operator/pkg/version.buildDate="2023-07-31T12:18:10Z" " github.com/openshift/csi-driver-manila-operator/cmd/csi-driver-manila-operator
```

Reply:
Looks like it's not working. Here are the logs we see on mecha-central where the job ran:
```
/var/log/containers/httpd/manila-api/manila_wsgi_access.log.4.gz:192.168.24.1 - - [28/Jul/2023:18:20:31 +0000] "GET /v2/7dbb1674a36d4954a9e2b20c25f9b770/types HTTP/1.1" 200 210 "-" "csi-driver-manila-operator/ gophercloud/2.0.0"
```

Looking at the [container file](https://github.com/openshift/csi-driver-manila-operator/blob/master/build/Dockerfile.openshift#L4), we don't use `make build`.

Reply:
This should be better with https://github.com/openshift/csi-driver-manila-operator/pull/194